public/static/settings.js

```diff
@@ -0,0 +1,83 @@
+// This function handles the toggling of selections of all upstream search engines 
+// options in the settings page under the tab engines.
+function toggleAllSelection() {
+  document
+    .querySelectorAll('.engine')
+    .forEach(
+      (engine_checkbox) =>
+        (engine_checkbox.checked =
+          document.querySelector('.select_all').checked)
+    )
+}
```

Comment:
The first function, `toggleAllSelection`, uses `querySelectorAll` which can be slow if there are many elements to search through. It also uses `forEach` which can be slow if there are many elements to iterate through. A more performant solution would be to use a `for loop` instead of `forEach`.

```suggestion
function toggleAllSelection() {
  const engineCheckboxes = document.querySelectorAll('.engine');
  const selectAllCheckbox = document.querySelector('.select_all');
  for (let i = 0; i < engineCheckboxes.length; i++) {
    engineCheckboxes[i].checked = selectAllCheckbox.checked;
  }
}
```